From dadb90c583d0ecf6569bee59261cd3587dce8b7c Mon Sep 17 00:00:00 2001
From: Javier Celaya <javier.celaya@flexvdi.com>
Date: Tue, 10 Jan 2017 17:33:45 +0100
Subject: [PATCH] Use alsasink instead of autoaudiosink

---
 src/spice-gstaudio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/spice-gstaudio.c b/src/spice-gstaudio.c
index dc1d525..d1014f8 100644
--- a/src/spice-gstaudio.c
+++ b/src/spice-gstaudio.c
@@ -317,7 +317,7 @@ static void playback_start(SpicePlaybackChannel *channel, gint format, gint chan
         gchar *pipeline = g_strdup (g_getenv("SPICE_GST_AUDIOSINK"));
         if (pipeline == NULL)
             pipeline = g_strdup_printf("appsrc is-live=1 do-timestamp=0 caps=\"%s\" name=\"appsrc\" ! queue ! "
-                                       "audioconvert ! audioresample ! autoaudiosink name=\"audiosink\"", audio_caps);
+                                       "audioconvert ! audioresample ! alsasink name=\"audiosink\"", audio_caps);
         SPICE_DEBUG("audio pipeline: %s", pipeline);
         p->playback.pipe = gst_parse_launch(pipeline, &error);
         if (error != NULL) {
-- 
2.10.2

