From e5e68f0eb9b16b1dc6320416498ef473d75c15db Mon Sep 17 00:00:00 2001
From: "Sergio L. Pascual" <sergiolpascual@flexvdi.com>
Date: Mon, 5 Oct 2015 12:44:38 +0200
Subject: [PATCH 3/3] Fix write_int64 macro for ARMLE.

---
 spice-common/common/marshaller.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/spice-common/common/marshaller.c b/spice-common/common/marshaller.c
index 9da7453..62ffc78 100644
--- a/spice-common/common/marshaller.c
+++ b/spice-common/common/marshaller.c
@@ -52,8 +52,8 @@
 #if defined(__arm__) || defined(_M_ARM)
 // In ARM, there may be problems with unaligned accesses in 64 bit values
 #warning Compiling for little-endian ARM architecture
-#define write_int64(ptr,v) do { *((uint32_t *)(ptr)) = *((uint32_t *)(&v)); \
-    *((uint32_t *)(ptr)+1) = *((uint32_t *)(&v)+1); } while(0)
+#define write_int64(ptr,v) do { *((uint32_t *)(ptr)) = (char)(v & 0xFFFFFFFF); \
+    *((uint32_t *)(ptr)+1) = (char)(v >> 32); } while(0)
 #define write_uint64(ptr,v) write_int64(ptr,v)
 #else
 #define write_int64(ptr,v) (*((int64_t *)(ptr)) = v)
-- 
2.3.8 (Apple Git-58)

