From 351ab22e520adc23fdb2e4cd40708b2632c0d706 Mon Sep 17 00:00:00 2001
From: "Sergio L. Pascual" <sergiolpascual@flexvdi.com>
Date: Tue, 17 May 2016 12:36:59 +0200
Subject: [PATCH] spice-channel.c: Fix alignment issues on ARMv7

---
 gtk/spice-channel.c | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/gtk/spice-channel.c b/gtk/spice-channel.c
index a15708f..d0e693f 100644
--- a/gtk/spice-channel.c
+++ b/gtk/spice-channel.c
@@ -1148,7 +1148,7 @@ static void spice_channel_send_link(SpiceChannel *channel)
 {
     SpiceChannelPrivate *c = channel->priv;
     uint8_t *buffer, *p;
-    int protocol, i;
+    int protocol, i, j;
 
     c->link_hdr.magic = SPICE_MAGIC;
     c->link_hdr.size = sizeof(c->link_msg);
@@ -1189,12 +1189,18 @@ static void spice_channel_send_link(SpiceChannel *channel)
     memcpy(p, &c->link_msg, sizeof(c->link_msg)); p += sizeof(c->link_msg);
 
     for (i = 0; i < c->common_caps->len; i++) {
-        *(uint32_t *)p = g_array_index(c->common_caps, uint32_t, i);
-        p += sizeof(uint32_t);
+        uint8_t *tmp = (uint8_t *) &g_array_index(c->common_caps, uint32_t, i);
+        for (j = 0; j < 4; j++) {
+            *p = tmp[j];
+            p++;
+        }
     }
     for (i = 0; i < c->caps->len; i++) {
-        *(uint32_t *)p = g_array_index(c->caps, uint32_t, i);
-        p += sizeof(uint32_t);
+        uint8_t *tmp = (uint8_t *) &g_array_index(c->caps, uint32_t, i);
+        for (j = 0; j < 4; j++) {
+            *p = tmp[j];
+            p++;
+        }
     }
     CHANNEL_DEBUG(channel, "channel type %d id %d num common caps %d num caps %d",
                   c->link_msg.channel_type,
-- 
2.7.4 (Apple Git-66)

